processing/src/main/java/org/apache/druid/segment/nested/NestedFieldLiteralDictionaryEncodedColumn.java

```diff
@@ -156,7 +156,11 @@ public String lookupName(int id)
   @Override
   public int lookupId(String name)
   {
-    return dictionary.indexOf(getIdFromGlobalDictionary(name));
+    final int globalId = getIdFromGlobalDictionary(name);
+    if (globalId < 0) {
+      return -1;
+    }
```

Comment:
Users of this aren't going to rely on the whole "here's the spot to insert" semantics, right?

Reply:
nope, dim selector doesn't have that contract since we pushed all of that business inside `BitmapColumnIndex`